src/main/java/org/broadinstitute/hellbender/engine/TwoPassVariantWalker.java

```diff
@@ -0,0 +1,90 @@
+package org.broadinstitute.hellbender.engine;
+
+import htsjdk.variant.variantcontext.VariantContext;
+import org.broadinstitute.hellbender.engine.filters.CountingReadFilter;
+import org.broadinstitute.hellbender.engine.filters.VariantFilter;
+import org.broadinstitute.hellbender.utils.SimpleInterval;
+
+import java.util.stream.StreamSupport;
+
+/**
+ * A VariantWalker that makes two passes through the variants.
+ * This allows the user to store internal states during the first pass, which the user can then
+ * process and access during the second pass
+ **/
+public abstract class TwoPassVariantWalker extends VariantWalker {
+    /**
+     * Overrides the default, single-pass traversal framework of {@link VariantWalkerBase}
+     */
+    @Override
+    public void traverse(){
+        final VariantFilter variantContextFilter = makeVariantFilter();
+        final CountingReadFilter readFilter = makeReadFilter();
+
+        // First pass through the variant
+        traverseVariants(variantContextFilter, readFilter, this::firstPassApply);
+        logger.info("Finished first pass through the variants");
+
+        // Process the data accumulated during the first pass
+        afterFirstPass();
+
+        // Second pass
+        logger.info("Starting second pass through the variants");
+        traverseVariants(variantContextFilter, readFilter, this::secondPassApply);
+    }
+
+    /**
+     *
+     * First pass through the variants. The user may store data in instance variables of the walker as you go
+     * and process them in {@link #afterFirstPass} before making a second pass
+     *
+     * @param variant A variant record in a vcf
+     * @param readsContext Reads overlapping the current variant. Will be empty if a read source (e.g. bam) isn't provided
+     * @param referenceContext Reference bases spanning the current variant
+     * @param featureContext A vcf record overlapping the current variant from an auxiliary data source (e.g. gnomad)
```

Comment:
`featureContext` supports more than just VCF records as auxiliary inputs -- any tribble-supported format can be accepted. I'd just change `vcf record` to `record`

Reply:
done